Source/WebCore/platform/graphics/ca/GraphicsLayerCA.cpp

```diff
@@ -1287,6 +1294,23 @@ void GraphicsLayerCA::setContentsToPlatformLayerHost(LayerHostingContextIdentifi
     noteLayerPropertyChanged(ContentsPlatformLayerChanged);
 }
 
+void GraphicsLayerCA::setContentsToVideoElement(HTMLVideoElement& videoElement, ContentsLayerPurpose purpose)
```

Comment:
The way to fix the layering violation would be to call something on the HTMLVideoElement that sets up this GraphicsLayer.